### PR TITLE
Update serial

### PR DESCRIPTION
**Pull request by Sebdar**

### CODE REVIEW

**Comment by Sebdar** [comment]
Protocol (as it it currently defined) is fully implemented. Waiting for reviews & tests with the actual Board A to merge (dont merge now!)

File: doc/serial.md [hunk 1]
```diff
@@ -42,3 +43,172 @@ sudo systemctl disable nvgetty
...
+
+8 bits 230400 baud 1 stop bit & no parity bit
+
+## Protocol
+
+General message layout
+
+|   SOF  | `cmd_id`  | `data_len` | `data ` |
+|--------|-----------|------------|---------|
+| 1 byte |  1 byte  |  1 byte    |         |
```
Comment (FlowSs) [inline doc/serial.md:64]: `cmd_id` was 2 bytes before, just to make sure the change to 1 in the doc is voluntary

File: doc/serial.md [hunk 2]
```diff
@@ -42,3 +43,172 @@ sudo systemctl disable nvgetty
...
+
+| Code    |     Description      |  `data_len` |
+|---------|----------------------|-------------|
+| `0x10`  | Turret target        | 4           |
+| `0x11`  | Movements            | 6           |
+| `0x12`  | Shoot order          | 0 (no data) |
+
+## Commands
+
+### Game status
```
Comment (FlowSs) [inline doc/serial.md:88]: 3v3 can also have 2 STD in place of a HRO or STY (if I remember correctly), is it accounted for?

**Comment by FlowSs** [comment]
Just added a few comments, for the technical stuff I trust everything is good :P When should we test with the boardA?

**Comment by FreshOlives** [comment]
@FlowSs I'm pretty busy this week, but next week / this weekend I'll go over my other PR and hopefully have something ready to go before the end of the week

**Comment by Sebdar** [comment]
Tested successfully with the Board A on Feb 27. A few tweaks but it works!

File: doc/serial.md (hunk 1 above)
Comment (FreshOlives) [reply doc/serial.md:64]: It's been implemented as 1 byte on the cs side as well, so I think we're good with this

**Comment by Sebdar** [comment]
@FlowSs Could you please approve so I can merge it?

**Comment by FlowSs** [comment]
Looks okay to me. And if you tested it then all good. Just one question: you flip a lot between uint and int, won't it cause any problem down the line? Like your theta is an int but your delta is uint

**Comment by Sebdar** [comment]
No problem so far since the size is still 16 bits, as long as it's coherently interpreted by the both of us (ahem fixed after testing it)

**Comment by FlowSs** [comment]
Perfect then

**Review by FlowSs — APPROVED** [review]

**Pull request merged**